### DISCUSSION TITLE
Prettylights Syntax Highlighting

### DISCUSSION THREAD

**Discussion by dominicegginton** [primer/github-vscode-theme#147] [Ideas]
Hi @simurai & contributors

I have opened this discussion to get everyone's opinion on the themes syntax highlighting. I am about to submit a PR that changes the hardcoded values for syntax highlighting to primer's prettylights syntax values available through [primer/primitives](https://github.com/primer/primitives). 

- Should we strictly follow GitHub.com's syntax highlighting? (This would provide consistency between VSCode, GitHub.com & Codespaces)
- Should we provide extra syntax highlighting functionality that GitHub.com does not provide? (E.G. #145, #129, #92 etc...)

**Comment by Amorpheuz** [comment]
Personally, I would love to see extra syntax highlighting to make the themes better and more usable in comparison to other well-fleshed out themes for all the color schemes under the extension.

Maybe, a good middle ground for either option could be that `GitHub Light`, `GitHub Dark` get better/extra highlighting functionality while the ones that are named same as the GitHub theme stick to their true essence of matching `GitHub.com`?

**Reply by dominicegginton** [reply]
Thanks for the feedback 😄

**Reply by simurai** [reply]
> Maybe, a good middle ground for either option could be that `GitHub Light`, `GitHub Dark` get better/extra highlighting functionality while the ones that are named same as the GitHub theme stick to their true essence of matching `GitHub.com`?

That's a good point. The `GitHub Light`, `GitHub Dark` should now have less pressure to mimic github.com.

**Comment by simurai** [comment]
This came up a few times in the past. I can see benefits for both:

> strictly follow GitHub.com's syntax highlighting

This makes it easy to decide and would keep things as consistent as possible.

> extra syntax highlighting functionality

You could argue that GitHub.com's syntax highlighting is too simple for an editor and mostly used to read code but not to edit code. Also the current setup makes it hard to really keep things consistent because different libraries get used:

- Prettylights when reading code
- Codemirror when editing code (single file editor)
- VS Code when editing with Codespaces

So might be fine the VS Code/Codespaces add some extra highlighting.

**Comment by sillypoise** [comment]
Don't know the state of this idea right now but I love the theme it's very well balanced. I am +1 for adding better syntax highlighting when editing. Cheers.

**Comment by dominicegginton** [comment]
@simurai if we add extra syntax highlighting features available for us in VSCode (for example: #145) could these features be implemented into GitHub's prettylights syntax for the web in order to keep consistency? I understand this could add work for many busy teams @GitHub but would be awsome 😄

**Reply by simurai** [reply]
That would be great. Turns out github.com uses two syntax highlighting. https://github.com/tree-sitter for a few popular languages and prettylights for the rest. Not sure how actively tree-sitter is maintained. 🤔

**Reply by dominicegginton** [reply]
Many thanks for the recourse links, I will have a look into this when I have time 😀
